lib/Twig/Test/IntegrationTestCase.php

```diff
@@ -17,27 +17,52 @@
  */
 abstract class Twig_Test_IntegrationTestCase extends PHPUnit_Framework_TestCase
 {
-    abstract protected function getExtensions();
+    /**
+     * @return string
+     */
     abstract protected function getFixturesDir();
 
     /**
-     * @dataProvider getTests
+     * @return Twig_ExtensionInterface[]
      */
-    public function testIntegration($file, $message, $condition, $templates, $exception, $outputs)
+    protected function getExtensions()
     {
-        $this->doIntegrationTest($file, $message, $condition, $templates, $exception, $outputs);
+        return array();
+    }
+
+    /**
+     * @return Twig_SimpleFilter[]
+     */
+    protected function getFilters()
+    {
+        return array();
     }
 
     /**
-     * @dataProvider getLegacyTests
-     * @group legacy
+     * @return Twig_SimpleFunction[]
      */
-    public function testLegacyIntegration($file, $message, $condition, $templates, $exception, $outputs)
```

Comment:
well, to me, it is, because it makes it very easy to handle things. If you handle it in a separate testcase, it means you have to register it twice and separate the storage of legacy and non-legacy fixtures.

The support of legacy integration tests is also part of the extension. Libraries extending Twig are also allowed to deprecate part of their own API while keeping it tested